services/provider/server/cephblockpool.go

```diff
@@ -0,0 +1,126 @@
+package server
+
+import (
+	"context"
+	"crypto/md5"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"slices"
+
+	rookCephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/ptr"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+)
+
+type cephBlockPoolManager struct {
+	client    client.Client
+	namespace string
+}
+
+func newCephBlockPoolManager(cl client.Client, namespace string) (*cephBlockPoolManager, error) {
+	return &cephBlockPoolManager{
+		client:    cl,
+		namespace: namespace,
+	}, nil
+}
+
+func (c *cephBlockPoolManager) EnableBlockPoolMirroring(ctx context.Context, blockPoolName string) error {
+
+	cephBlockPool, err := c.get(ctx, blockPoolName)
```

Comment:
you don't need to get the resource, just fill the name, namespace and mutate the desired state in createorupdate.

Reply:
But that would mean that if a blockpool doesn't exist, it will create one. We don't want to create the blockpool

Reply:
I see change made, can be resolved.